code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -437,12 +437,16 @@
 	taste_description = "ash"
 
 /datum/reagent/medicine/charcoal/on_mob_life(mob/living/carbon/M)
+	M.adjust_disgust(20)
 	M.adjustToxLoss(-2*REM, 0)
 	. = 1
 	for(var/datum/reagent/R in M.reagents.reagent_list)
 		if(R != src)
 			M.reagents.remove_reagent(R.type,1)
 	..()
+/datum/reagent/medicine/charcoal/reaction_mob(mob/living/M, methods=TOUCH)
+	if(methods & INGEST)
+		to_chat(M, span_notice("You feel a bit queasy, you shouldn't have too much of this"))
```

Comment:
You need `return ..()` after this line to not override the normal `reaction_mob()` function

Reply:
whoops thanks